src/d3d11/d3d11_context.cpp

```diff
@@ -3624,28 +3624,34 @@ namespace dxvk {
           UINT                              NumBuffers,
           ID3D11Buffer* const*              ppConstantBuffers) {
     uint32_t slotId = computeConstantBufferBinding(ShaderStage, StartSlot);
-    
+
     for (uint32_t i = 0; i < NumBuffers; i++) {
       auto newBuffer = static_cast<D3D11Buffer*>(ppConstantBuffers[i]);
-      
-      UINT constantBound = 0;
-      
+
+      UINT constantCount = 0;
+
       if (likely(newBuffer != nullptr))
-        constantBound = newBuffer->Desc()->ByteWidth / 16;
-      
+        constantCount = newBuffer->Desc()->ByteWidth / 16;
+
       if (Bindings[StartSlot + i].buffer         != newBuffer
-       || Bindings[StartSlot + i].constantBound  != constantBound) {
+       || Bindings[StartSlot + i].constantCount  != constantCount) {
         Bindings[StartSlot + i].buffer         = newBuffer;
         Bindings[StartSlot + i].constantOffset = 0;
-        Bindings[StartSlot + i].constantCount  = constantBound;
-        Bindings[StartSlot + i].constantBound  = constantBound;
-        
-        BindConstantBuffer(slotId + i, newBuffer);
+        Bindings[StartSlot + i].constantCount  = constantCount;
+
+        if (likely(constantCount <= UINT(D3D11_REQ_CONSTANT_BUFFER_ELEMENT_COUNT))) {
```

Comment:
Why do we need this `if`? We should be able to just set `constantCount` to the proper value in line 3634.

Reply:
We need the if because otherwise it calls BindConstantBuffer (without the 1) which binds the entire slice.

Reply:
Just always call 1 instead?

Anyway, gonna do some testing since I want to make sure this is even the right thing to do.

Reply:
Updated the PR to just use BindConstantBuffer1.